### DISCUSSION TITLE
requirediscovered="false" does not work

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#14315] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

`requirediscovered="false"` is supposed to allow locations changing in the fog of war, but it doesnt

In our mod you can see locations failing to turn in the fog of war as well as barely visible next to the fog of war
Only after you move near or past the fog of war do the Changetos trigger

_Lairs (the egg locations with a mission on it) turn adjacent locations into Abandoned/None, and then turns Abandoned/None into "Outerlair" (broken eggs icon)._

https://cdn.discordapp.com/attachments/1200592925382422530/1264158049527005235/Desktop_2023.09.10_-_17.38.09.06_52.mp4?ex=669cda6d&is=669b88ed&hm=eee0ca2069ef63af04ca5185e36cf350315b977bdfc3847f759bb0624de7b2eb&

```xml
      <ChangeTo type="Abandoned" disallowedproximity="1" locationtypechangecooldown="2" messagetag="locationchange.outpost.changeto.abandoned" requirediscovered="false">
        <Requirement requiredlocations="Lair,LairMudraptor,LairThresher" probability="1" requireddurationrange="1,2" requiredproximity="1" requirediscovered="false" />
      </ChangeTo>
````

### Reproduction steps

1. Use Dynamic Europa
https://steamcommunity.com/sharedfiles/filedetails/?id=2532991202

2. Start campaign
3. Debugdraw
4. Togglecampaignteleport
5. Find a lair on the map in the fog of war (there's one per biome)
6. Teleport between locations to trigger map changes a few times
7. Lair doesnt change
8. Teleport near the lair, locations near lair start changing

xml of locationtypes
[LocationTypes.zip](https://github.com/user-attachments/files/16319748/LocationTypes.zip)

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
> requirediscovered="false" is supposed to allow locations changing in the fog of war, but it doesnt

Actually "discovering" seems to mean that the location needs to have been visited at least once.

**Reply by ubertpendragon** [reply]
oh ok. 

It's a "locations not turning bug" then
